baseapp/baseapp.go

```diff
@@ -629,6 +702,12 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte, tx sdk.Tx) (result sdk
 			}
 		}
 
+		// consume block gas whether panic or not.
```

Comment:
What happens if this panics again? Won't `DeliverTx` panic, whereas we want to just return an error?

Reply:
What would specifically panic again?

Reply:
The `ConsumeGas` call in the recover handler here.

Reply:
If this panics it will be caught by the recover (just above this comment) - this comment simply notes that even if this consume gas panics, it will still affect the state of the BlockGasMeter which is important for ignoring future transactions... updating this comment to reflect this